### PR TITLE
vbam: add wxWidgets GUI

### PR DESCRIPTION
**Pull request by aske**
###### Motivation for this change
Add a wxWidgets GUI with a working GBA link.
Had to update `vbam` to the latest git revision, because GBA link wasn't building on the latest 2.1.4 release.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by Lassulus** [comment]
I would prefer to wait for a new stable release

**Comment by NorfairKing** [comment]
Just FYI: I got this building with `wxGTK31` and the latest release instead. Thanks for opening this PR, even if it didn't get merged.

File: pkgs/misc/emulators/vbam/default.nix [hunk 1]
```diff
@@ -11,18 +12,23 @@
...
 , zlib
+, enableWx ? false
+, gtk2
+, wxGTK
 }:
 
 stdenv.mkDerivation rec {
   pname = "visualboyadvance-m";
-  version = "2.1.4";
+  version = "2019-12-26";
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/vbam/default.nix:22]:
```suggestion
  version = "unstable-2019-12-26";
```

File: pkgs/misc/emulators/vbam/default.nix [hunk 2]
```diff
@@ -35,23 +41,25 @@ stdenv.mkDerivation rec {
...
     "-DENABLE_LINK='true'"
     "-DSYSCONFDIR=etc"
-    "-DENABLE_WX='false'"
-    "-DENABLE_SDL='true'"
-  ];
+  ] ++ (if !enableWx
+        then [ "-DENABLE_WX='false'"
+               "-DENABLE_SDL='true'"
+             ]
+        else [ "-DENABLE_WX='true'" ]);
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/vbam/default.nix:55]:
```suggestion
  ] ++ (if !enableWx
    then [ "-DENABLE_WX='false'"
      "-DENABLE_SDL='true'"
    ] else [
      "-DENABLE_WX='true'" 
    ]);
```

File: pkgs/misc/emulators/vbam/default.nix [hunk 3]
```diff
@@ -35,23 +41,25 @@ stdenv.mkDerivation rec {
...
 
-  meta =  with stdenv.lib; {
+  meta =  with lib; {
     description = "A merge of the original Visual Boy Advance forks";
     license = licenses.gpl2;
     maintainers = with maintainers; [ lassulus ];
     homepage = http://vba-m.com/;
-    platforms = stdenv.lib.platforms.linux;
+    platforms = lib.platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/vbam/default.nix:62]:
```suggestion
    platforms = platforms.linux;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Closing as essentially dead.

**Pull request closed without merging**